example/concurrent/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/RobotsAndPencils/buford/certificate"
+	"github.com/RobotsAndPencils/buford/payload"
+	"github.com/RobotsAndPencils/buford/push"
+)
+
+func main() {
+	var deviceToken, filename, password, environment string
+	var workers uint
+	var number int
+
+	flag.StringVar(&deviceToken, "d", "", "Device token")
+	flag.StringVar(&filename, "c", "", "Path to p12 certificate file")
+	flag.StringVar(&password, "p", "", "Password for p12 file")
+	flag.StringVar(&environment, "e", "development", "Environment")
+	flag.UintVar(&workers, "w", 20, "Workers to send notifications")
+	flag.IntVar(&number, "n", 100, "Number of notifications to send")
+	flag.Parse()
+
+	log.SetFlags(log.Ltime | log.Lmicroseconds)
+
+	cert, err := certificate.Load(filename, password)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// establish a connection to Apple
+	client, err := push.NewClient(cert)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	service := push.NewService(client, push.Development, workers)
+	if environment == "production" {
+		service.Host = push.Production
+	}
+
+	// wait group to wait for all responses
+	var wg sync.WaitGroup
+
+	// process responses
```

Comment:
It's important that this is setup prior to sending notifications, otherwise there won't be room in the responses channel. Document this more thoroughly.